libs/gi/db/src/Database/DataManagers/OptConfigDataManager.ts

```diff
@@ -333,6 +324,35 @@ function mapFilterInplace<T>(xs: T[], f: (x: T) => T | undefined): T[] {
   return reuse ? xs : a
 }
 
+/**
+ * Like `[...new Set(xs)].filter(f)` but if the elements are all unique and `f`
+ * returns true, the original array is returned
+ *
+ * If`xs` is not an array, `[]` is returned
+ *
+ * React compares objects by pointer equality, so `[...new Set(xs)].filter(f)`
+ * always causes a rerender
+ */
+function uniqueFilterInplace<T>(xs: T[], f: (x: T) => boolean): T[] {
```

Comment:
I think it may make sense to use `React.memo`'s `arePropsEqual` to do a JSON string comparison on the UI update side, instead of "conserving references" in database (which kind of complicates the db logic).